src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+    results = dict(task.iter_retcodes())
+    if 0 in results:
+        for output, nodelist in task.iter_buffers(results[0]):
+            output_lines = [line.decode("utf-8") for line in output]
+            nodeset = NodeSet.fromlist(nodelist)
+            for line in output_lines:
+                try:
+                    speed = int(line.strip())
+                except IOError as io_error:
+                    # KVM/Qemu/libvirt returns an EINVAL
+                    if io_error.errno == errno.EINVAL:
+                        speed = 1000
+                except ValueError:
+                    # Any line not containing a speed (integer)
+                    continue
+                if speed not in interface_speeds:
+                    interface_speeds[speed] = NodeSet()
+                interface_speeds[speed].update(nodeset)
+
+    return interface_speeds
+
+
+def get_interface_numa_node(hosts, interface, verbose=True):
+    """Get the NUMA node ID of this network interface on each host.
+
+    Args:
+        hosts (NodeSet): hosts on which to detect the NUMA node
+        interface (str): interface for which to obtain the NUMA node
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of NUMA node ID keys and NodeSet values on which they were detected
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    command = "cat {}".format(os.path.join(net_path, interface, "device", "numa_node"))
+    task = run_task(hosts, command, verbose=verbose)
+    if verbose:
+        display_task(task)
+
+    # Populate a dictionary of numa node IDs with a NodSet of hosts on which it was detected
+    numa_nodes = {}
+    results = dict(task.iter_retcodes())
+    if 0 in results:
+        for output, nodelist in task.iter_buffers(results[0]):
+            output_lines = [line.decode("utf-8") for line in output]
+            nodeset = NodeSet.fromlist(nodelist)
+            for line in output_lines:
+                try:
+                    numa_node = int(line.strip())
+                except ValueError:
+                    continue
+                if numa_node not in numa_nodes:
+                    numa_nodes[numa_node] = NodeSet()
+                numa_nodes[numa_node].update(nodeset)
+
+    return numa_nodes
+
+
+def get_interface_ib_name(hosts, interface, verbose=True):
+    """Get the InfiniBand name of this network interface on each host.
+
+    Args:
+        hosts (NodeSet): hosts on which to detect the InfiniBand name
+        interface (str): interface for which to obtain the InfiniBand name
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of InfiniBand name keys and NodeSet values on which they were detected
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    command = "ls -1 {}".format(os.path.join(net_path, interface, "device", "infiniband"))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string